src/components/segmentedControl/index.tsx

```diff
@@ -170,58 +191,75 @@ const SegmentedControl = (props: SegmentedControlProps) => {
 
   const renderSegments = () =>
     _.map(segments, (_value, index) => {
+      const isLastSegment = index + 1 === segments?.length;
       return (
-        <Segment
-          key={index}
-          onLayout={onLayout}
-          index={index}
-          onPress={onSegmentPress}
-          selectedIndex={animatedSelectedIndex}
-          activeColor={activeColor}
-          inactiveColor={inactiveColor}
-          style={segmentsStyleProp}
-          segmentLabelStyle={segmentLabelStyle}
-          {...segments?.[index]}
-          testID={testID}
-        />
+        <>
+          <Segment
+            key={index}
+            onLayout={onLayout}
+            index={index}
+            onPress={onSegmentPress}
+            selectedIndex={animatedSelectedIndex}
+            activeColor={activeColor}
+            inactiveColor={inactiveColor}
+            style={[segmentsStyleProp]}
+            segmentLabelStyle={segmentLabelStyle}
+            iconTintColor={iconTintColor}
+            {...segments?.[index]}
+            testID={testID}
+          />
+          {!isLastSegment && (
+            <View
+              key={`segment.divider-${index}`}
+              width={segmentDividerWidth}
+              height={'100%'}
+              style={{backgroundColor: segmentDividerColor}}
+            />
+          )}
+        </>
       );
     });
-
   return (
     <View style={containerStyle} testID={testID}>
       <View row center style={[styles.container, style, {borderRadius, backgroundColor}]}>
+        <Reanimated.View
+          style={[
+            styles.selectedSegment,
+            {
+              borderRadius,
+              backgroundColor: activeBackgroundColor
+            },
+            animatedStyle
+          ]}
+        />
+        {renderSegments()}
         <Reanimated.View
           style={[
             styles.selectedSegment,
             {
               borderColor: outlineColor,
               borderRadius,
-              backgroundColor: activeBackgroundColor,
               borderWidth: outlineWidth
             },
             animatedStyle
           ]}
         />
-        {renderSegments()}
       </View>
     </View>
   );
-};
+}) as SegmentedControlComponent;
```

Comment:
Why is that for?

Reply:
This is so I could put the preset property on the component.